### ISSUE TITLE
CST: Fix linting issues

### ISSUE THREAD

**Issue by Mottie** [department-of-veterans-affairs/va.gov-team#36054]
## Description

Accessibility linting rules are being enabled, and 11 linting errors popped up within the claim status tool

<details><summary>Linting errors</summary>

- [x] `src/applications/claims-status/components/AddFilesForm.jsx`
  166:13  error  Anchor used as a button. Anchors are primarily expected to navigate. Use the button element instead. Learn more: https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/anchor-is-valid.md  jsx-a11y/anchor-is-valid

- [x] `src/applications/claims-status/components/ClaimPhase.jsx`
  183:7  error  Visible, non-interactive elements with click handlers must have at least one keyboard listener  jsx-a11y/click-events-have-key-events

- [x] `src/applications/claims-status/components/AppealEventItem.jsx`
  31:7  error  Visible, non-interactive elements with click handlers must have at least one keyboard listener  jsx-a11y/click-events-have-key-events
:arrow_right: **Notes:** This is `<li>` with an `onClick` that reveals a description. This might require an update to the UI to make accessible.
:arrow_right: Dead component - satisfy linter only

- [x] `src/applications/claims-status/components/ClaimSyncWarning.jsx`
  21:11  error  Anchor used as a button. Anchors are primarily expected to navigate. Use the button element instead. Learn more: https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/anchor-is-valid.md  jsx-a11y/anchor-is-valid
:arrow_right: **Notes:** This is an un-classed `<a>` element, which means it looks like a link. It should be a `<button>` and look like a button. Changing the tag will change the UI. It should be viewed in the browser to assess for breaking the design or if it introduces visual tension. Also might need to determine which button variant to use.

- [x] `src/applications/claims-status/components/UploadStatus.jsx`
  17:11  error  Anchor used as a button. Anchors are primarily expected to navigate. Use the button element instead. Learn more: https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/anchor-is-valid.md  jsx-a11y/anchor-is-valid
:arrow_right: **Notes:** This is an un-classed `<a>` element, which means it looks like a link. It should be a `<button>` and look like a button. Changing the tag will change the UI. It should be viewed in the browser to assess for breaking the design or if it introduces visual tension. Also might need to determine which button variant to use.

- [x] `src/applications/claims-status/components/appeals-v2/Expander.jsx`
  26:5  error  Visible, non-interactive elements with click handlers must have at least one keyboard listener  jsx-a11y/click-events-have-key-events
:arrow_right: **Notes:** Possibly resolved? There's eslint disable comment here. Looks the component has a button in it and `<li>` does not use an `onClick` handler.
✅ This issue was resolved. No `eslint` complaints on the page.

- [x] `src/applications/claims-status/containers/AskVAPage.jsx`
  102:17  error  Visible, non-interactive elements with click handlers must have at least one keyboard listener jsx-a11y/click-events-have-key-events
:arrow_right: **Notes:** ~~Might be an swap from `<a>` to `<button>` because the link already has class name `usa-button-secondary`.~~ `usa-button-secondary` used on `<a>` does not give the link button styling.
  102:17  error  Anchor used as a button. Anchors are primarily expected to navigate. Use the button element instead. Learn more: https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/anchor-is-valid.md  jsx-a11y/anchor-is-valid
:arrow_right: **Notes:** Redundant error. This is because the `<a>` does not have an `href` attribute.
  102:17  error  Static HTML elements with event handlers require a role                                                                                                                                                       jsx-a11y/no-static-element-interactions
:arrow_right: **Notes:** Redundant error. This is because the `<a>` does not have an `href` attribute.

- [x] `src/applications/claims-status/containers/DocumentRequestPage.jsx`
  147:17  error  The href attribute requires a valid value to be accessible. Provide a valid, navigable address as the href value. If you cannot provide a valid href, but still need the element to resemble a link, use a button and change it with appropriate styles. Learn more: https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/anchor-is-valid.md  jsx-a11y/anchor-is-valid
:arrow_right: **Notes:** Potentially resolved. Uses `<Link to="">`
✅ This `eslint` complaint is no longer on the page.

- [x] `src/applications/claims-status/tests/components/Breadcrumbs.unit.spec.jsx`
  11:9  error  The href attribute requires a valid value to be accessible. Provide a valid, navigable address as the href value. If you cannot provide a valid href, but still need the element to resemble a link, use a button and change it with appropriate styles. Learn more: https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/anchor-is-valid.md  jsx-a11y/anchor-is-valid
:arrow_right: **Notes:** Just a test. Just change `href="#"` to `href="/foo"` and update the assertion.

~~`src/applications/claims-status/utils/appeal-helpers.jsx`~~
  351:15  error  The href attribute requires a valid value to be accessible. Provide a valid, navigable address as the href value. If you cannot provide a valid href, but still need the element to resemble a link, use a button and change it with appropriate styles. Learn more: https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/anchor-is-valid.md  jsx-a11y/anchor-is-valid
:arrow_right: **Notes:** Need to learn more about this, but the line is `<a href="#">Learn more about hearings</a>`. Suggest getting rid of the line and it's parent `<p>`. OR we need to know what the link is. The other "learn more" links in this helper are broken links.
:arrow_right: This is moved to a new issue: https://github.com/department-of-veterans-affairs/vets-website/issues/21233

</details>

## Tasks

- [x] Fix a11y linting errors in the claim status tool
- [x] Update tests as needed

## Definition of done

- [x] All tasks complete
- [x] All tests passing

**Comment by Mottie** [comment]
This work was only partially done this sprint. The delay to getting more done is 1) inability to view exact pages within the CST (need user or mock data to view) or 2) Discussion with design regarding the design considerations and/or alternatives of changing a link into a button.

**Comment by jaredcunha** [comment]
I've added some context on each linting error to the list.

**Comment by tameracorson** [comment]
@jaredcunha did you mean to drop this on health hub board?

**Comment by jaredcunha** [comment]
@tameracorson I did not - I wonder if I accidentally dragged or something?
